src/manager/TickManager.ts

```diff
@@ -1,46 +1,190 @@
-import JSBI from 'jsbi';
+import JSBI from "jsbi";
+import assert from "assert";
 import { Tick } from "../model/Tick";
 import { jsonMapMember, jsonObject } from "typedjson";
 
 @jsonObject
 export class TickManager {
   @jsonMapMember(Number, Tick, { name: "ticks_json" })
+  // key = Tick.tickIndex
   private ticks: Map<number, Tick>;
 
   constructor(ticks: Map<number, Tick> = new Map()) {
     this.ticks = ticks;
   }
 
   get(tickIndex: number): Tick {
-    return this.ticks.get(tickIndex) || new Tick();
+    if (this.ticks.has(tickIndex)) return this.ticks.get(tickIndex)!;
+    const newTick = new Tick(tickIndex);
+    this.set(newTick);
+    return newTick;
   }
 
-  set(tickIndex: number, tick: Tick) {
-    this.ticks.set(tickIndex, tick);
+  set(tick: Tick) {
+    this.ticks.set(tick.tickIndex, tick);
   }
 
-  flipTick(tickIndex: number, tickSpacing: number) {
-    // TODO
+  private nextInitializedTick(
+    sortedTicks: readonly Tick[],
+    tick: number,
+    leftToRight: boolean
+  ): Tick {
+    if (leftToRight) {
+      assert(!this.isBelowSmallest(sortedTicks, tick), "BELOW_SMALLEST");
+      if (this.isAtOrAboveLargest(sortedTicks, tick)) {
+        return sortedTicks[sortedTicks.length - 1];
+      }
+      const index = this.binarySearch(sortedTicks, tick);
+      return sortedTicks[index];
+    } else {
+      assert(
+        !this.isAtOrAboveLargest(sortedTicks, tick),
+        "AT_OR_ABOVE_LARGEST"
+      );
+      if (this.isBelowSmallest(sortedTicks, tick)) {
+        return sortedTicks[0];
+      }
+      const index = this.binarySearch(sortedTicks, tick);
+      return sortedTicks[index + 1];
+    }
   }
 
   getNextInitializedTick(
     tick: number,
     tickSpacing: number,
     leftToRight: boolean
```

Comment:
Sorry I mischanged the input parameter name again. Please change this to `lte`(as contract did) or `zeroForOne`(as context means).